mapper/orm/src/main/java/org/hibernate/search/mapper/orm/search/query/SearchQuery.java

```diff
@@ -72,19 +72,35 @@
 	 * <p>
 	 * The default is no limit.
 	 *
+	 * @param maxResults The maximum number of results to return. Must be positive or zero; {@code null} resets to the default value.
+	 * @return {@code this} for method chaining.
+	 */
+	SearchQuery<T> setMaxResults(Long maxResults);
+
+	/**
+	 * @deprecated Pass a {@code long} value and use {@link #setMaxResults(Long)} instead.
 	 * @param maxResults The maximum number of results to return. Must be positive or zero.
 	 * @return {@code this} for method chaining.
 	 */
+	@Deprecated
 	SearchQuery<T> setMaxResults(int maxResults);
 
 	/**
 	 * Set the offset of the first result returned by this query.
 	 * <p>
 	 * The default offset is {@code 0}.
 	 *
+	 * @param firstResult The offset of the first result. Must be positive or zero; {@code null} resets to the default value.
+	 * @return {@code this} for method chaining.
+	 */
+	SearchQuery<T> setFirstResult(Long firstResult);
+
+	/**
+	 * @deprecated Pass a {@code long} value and use {@link #setFirstResult(Long)} instead.
 	 * @param firstResult The offset of the first result. Must be positive or zero.
 	 * @return {@code this} for method chaining.
 	 */
+	@Deprecated
```

Comment:
Should we remove them instead? My rationale is that people will have a tendency to use them considering you don't need the `L` and they would have a warning.

Reply:
If I remove them, people trying to pass an `int` (either a literal without a trailing `L`, or a variable) will get a compilation error. There is no automatic boxing from `int` to `Long`, just from `long` to `Long`.

If I removed the `int` variants and changed the parameter type from `Long` to `long`, then users would benefit from automatic casting from `int` to `long`. But then they couldn't reset the value by passing `null` to `setFirstResult` or `setMaxResults` anymore. Which I suppose is fine, since they could pass `0` to `setFirstResult` and `Long.MAX_VALUE` to setMaxResults`?

Reply:
Well, what I mean is: either you support it and you don't have them deprecated or you don't.

But what you're saying in your first paragraph means they will end up with a deprecation warnings.

Reply:
Ok, so I suppose "them" meant the `@Deprecated` annotations, not the methods. Anyway... It's probably not relevant anymore if we switch to a `fetch(limit, offset)` method.

Reply:
I removed the deprecation, we now have two variants of each `fetch` method: one accepting longs and one accepting integers.